### PR TITLE
Remove the calls to realpath in FileIntegrity checks

### LINKED ISSUE #16551: File integrity gives bad advice on symlinks
I added symlinks to matomo.js and matomo.php because matomo is a keyword that is blocked by some blockers.

File integrity is correct in pointing out they should not be there, but the solution it proposes is wrong.
Screenshot 2020-10-08 at 13 56 40

Matomo 3.14.1

2020-10-10 - edit fixed typo

### PR DESCRIPTION
**Pull request by danielpunkass**
### Description:

If the FileIntegrity check discovers an unexpected symlink in the source tree, it causes Matomo to suggest to the user deleting the target of the symlink, instead of the symlink itself. At best this would be a nuisance, and at worst it could be catastrophic (if the user kept a symlink to files outside of Matomo's directory, for example). Fixes #16551.

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

**Comment by caddoo** [comment]
@danielpunkass Thanks a lot for your contribution, really appreciated 🎉 

I tested this locally and while it does work and achieve what is expected, we do lose a little user-friendliness with the messaging to our users.

Removing `realpath` means a relative filename is given which means the `rm` or `rm -rF` will only work when they are in their Matomo install directory whereas before they could be anywhere in their filesystem.

Maybe an idea would be to change it to use something like:
`htmlspecialchars(realpath(dirname($fileFoundNotExpected)). '/' .basename($fileFoundNotExpected))` 

This seems to get the symlink location with the full absolute path and the resulting `rm` command doesn't touch the source file.

@matomo-org/core-reviewers  any other opinions here?

File: core/FileIntegrity.php [hunk 1]
```diff
@@ -92,7 +92,7 @@ protected static function getMessagesDirectoriesFoundButNotExpected($messages)
 
             $directories = array();
             foreach ($directoriesFoundButNotExpected as $directoryFoundNotExpected) {
-                $directories[] = htmlspecialchars(realpath($directoryFoundNotExpected));
+                $directories[] = htmlspecialchars($directoryFoundNotExpected);
```
Comment (caddoo) [inline core/FileIntegrity.php]:
```suggestion
                $directories[] = htmlspecialchars(realpath(dirname($directoryFoundNotExpected)). DIRECTORY_SEPARATOR . basename($directoryFoundNotExpected));
```

File: core/FileIntegrity.php [hunk 2]
```diff
@@ -138,7 +138,7 @@ protected static function getMessagesFilesFoundButNotExpected($messages)
 
             $files = array();
             foreach ($filesFoundButNotExpected as $fileFoundNotExpected) {
-                $files[] = '"' . htmlspecialchars(realpath($fileFoundNotExpected)) . '"';
+                $files[] = '"' . htmlspecialchars($fileFoundNotExpected) . '"';
```
Comment (caddoo) [inline core/FileIntegrity.php]:
```suggestion
                $files[] = '"' . htmlspecialchars(realpath(dirname($fileFoundNotExpected)). DIRECTORY_SEPARATOR . basename($fileFoundNotExpected)) . '"';

```

**Review by caddoo — COMMENTED** [review]
@danielpunkass I don't have push access to your fork, but i've made some suggested changes, can you add them as commits and we can see how the tests react 😄

File: core/FileIntegrity.php (hunk 2 above)
Comment (danielpunkass) [reply core/FileIntegrity.php]: Thanks for the suggestions, and I'm sorry I somehow missed your earlier comment + suggestion! Thanks for taking the time to evaluate and consider a hopefully good balance between past behavior and avoiding the bug. I'll update the PR and as you say, see how it goes!

**Review by danielpunkass — COMMENTED** [review]
Incorporate changes suggested by @caddoo

**Comment by danielpunkass** [comment]
I will figure out how to quiet the style errors.

**Review by caddoo — APPROVED** [review]

**Pull request merged**